lib/model-access-wrapper.js

```diff
@@ -523,9 +523,12 @@ class ModelAccessWrapper extends CrispHooks {
 
 				// Check permissions
 				if (params.permissions) {
-					let grant = params.permissions.getTargetGrant(this.options.permissionsTarget, data);
-					grant.check('write');
-					grant.checkMask('writeMask', data);
+					// Test permissions against an update from a new doc to the passed data,
+					// which allows creating docs when the user does not have permissions for defaulted fields.
+					const initial = model.create();
+					const changedData = Update.createFromDiff(initial.data, data).$set;
```

Comment:
This is not a valid way of checking these permissions.  What if the update contains operators other than $set (which it often does)?

You need to use the `changedData` Update to get a list of modified fields.  I believe Update has a method called getUpdatedFields() or similar specifically for this purpose.  Then the permission grant has a method (I forget the name) that checks to ensure all fields in an array of fields can be accessed.

Check the `update()` method in this same file for how that handles permissions.  It may do it the same way that you need to do it here.